api/envoy/api/v2/core/protocol.proto

```diff
@@ -83,6 +84,10 @@ message Http1ProtocolOptions {
   // Describes how the keys for response headers should be formatted. By default, all header keys
   // are lower cased.
   HeaderKeyFormat header_key_format = 4;
+
+  // Disables encoding trailers for HTTP1. Setting this to true will drop the
+  // trailers during encoding.
+  bool disable_trailers = 5;
```

Comment:
@alyssawilk @PiotrSikora as long as we are adding this option, I'm tempted to actually flip the default, so make this `enable_trailers`. Basically no one wants this in the real world and I'm not sure it's worth the regression risk? WDYT?

Reply:
Why do you say that no one wants this in the real world?

Reply:
Because I have never seen an actual use case for this in practice, so I don't think it's a heavily requested feature. Given that, I'm wondering if it's better to make this opt-in vs. opt-out, though I don't feel super strongly about it.

Reply:
For open internet trailers are poorly supported, so I'd be inclined to have it off-by-default, but I have no idea what folks on the service mesh side are doing so I'm fine with whatever Matt thinks is best (which sounds like is default off?)

Orthogonal from this can we make sure the grpc web filter is fixed before this lands?

Reply:
OK let's flip the name of the config option to `enable_trailers` and default to off and then fix the gRPC web filter? Thank you!

/wait

Reply:
> Orthogonal from this can we make sure the grpc web filter is fixed before this lands?

Is this something this PR breaks or are you talking about a different issue outside this PR?

As for the default on/off. I'm fine either way, just let me know what you folks decide and I'll make the change :smile:

Reply:
@alyssawilk yeah, trailers are poorly supported on the open Internet, because proxies don't forward them by default, so it's a catch-22. We don't use them in service mesh, my comment was in service of the better HTTP support on the open Internet.

In any case, it's not a blocker, just a personal preference.

Reply:
This is something the PR makes worse.
the gRPC web filter takes the trailers and adds them to the request body.
Now that trailers are supported, I believe we should either clear the trailer fields as we add them to the body (even if trailers are configured on) because that's how gRPC web works.
I'd suggest waiting for @lizan correct behavior for gRPC web where HTTP/1.1 trailers are supported before you do any work though.

Reply:
Yes, the HTTP trailers should be removed after they are merged into HTTP body in gRPC-Web, otherwise we're going to end up with 2 sets of trailers, and possible duplicates in case of gRPC-Web -> gRPC conversion.

Reply:
@alyssawilk @PiotrSikora Since this flag is now default off, are we okay merging it in as is since the risk is a lot lower now.

Reply:
I think we should still strip them from gRPC-Web responses, but I'll defer that decision to the maintainers.

Reply:
If default behavior is unchanged by the PR, I wonder why gRPC-web filter / responses would be affected?

Reply:
Because if someone enables trailers, they will end up with trailers duplicated in 2 places. 

The fact that it's off by default doesn't change anything.

Reply:
I would suggest decouple them if possible. Existing gRPC-web users will not see any regression at least. If some one accidentally use gRPC-web and enable trailers, they still have a way to temporarily mitigate until gRPC-web filter is updated.

Reply:
The one thing we have to watch out for though is that this PR now decodes H1 trailers which is not behind a future flag. I'm not sure how this filter works so I'll have to invest some time investigating.

Reply:
I think both encoding and decoding headers should be config guarded.  Can we make that change as well? 
I'm fine with gRPC web fixes landing in a separate PR, but I'd prefer that PR land first if we split it out.

Reply:
Thanks! IMHO, this PR would affect all use cases that assume the trailers are dropped, gRPC-web is one of them. Even if we fix gRPC web first, the other use cases would face the same potential issue. That's why I think the PR and gRPC-web are orthogonal and independent and we shouldn't treat gRPC-web as a special case.

Reply:
@alyssawilk both decoding and encoding trailers are now config guarded. Please let me know if I missed anything else. With the config guard, I'm more confident that this is a lower risk.

>I'm fine with gRPC web fixes landing in a separate PR, but I'd prefer that PR land first if we split it out.

Are you talking about this PR landing first or the gRPC web fixes? Is there already an ongoing GRPC web fix pr going on?